### PR TITLE
[ENHANCEMENT] Support Delete Invoice and Get Invoice Item

### PR DESCRIPTION
**Pull request by mayanastasya**
Delete Invoice Item will return this error response:
```
{
  "errors"=>[
    { "Code" => "INVALID_VALUE", "Message" => "Only Cancelled invoices can be deleted." }
  ],
  "id" => "2c92a0aa6a2e994f016a2fc9df2c0c6b",
  "success" => false
}
```

### CODE REVIEW

File: lib/zuora/invoice.rb [hunk 1]
```diff
@@ -15,6 +15,14 @@ def find_by_account_id_base_resource_url(account_id)
         def find_by_id(invoice_id)
           Zuora.request(:get, object_url(invoice_id), {error_handler: Zuora::ErrorHandler::StatusChecker})
         end
+
+        def get_items(invoice_id)
```
Comment (ryancyq) [inline lib/zuora/invoice.rb]: How about
```rb
def invoice_items
    # Your code
end

Comment (mayanastasya) [reply lib/zuora/invoice.rb]: Not sure about method naming standards. @gherry ? Other methods use verbs tho (find, cancel, delete, renew).

Comment (ryancyq) [reply lib/zuora/invoice.rb]: i think usually verb is used for current object (in this case, `invoice`).
to list the relationships, usually use the relationships/collection name.

e.g. 
order.destroy
order.line_items.first.destroy

Comment (gherry) [reply lib/zuora/invoice.rb]: If we follow the current convention then we should have `Zuora::InvoiceItems` endpoint and then have a method called `find_by_invoice_id`

Comment (gherry) [reply lib/zuora/invoice.rb]: One thing that is similar to this is https://github.com/tradegecko/zuora-rest-ruby/blob/master/lib/zuora/invoice.rb#L7

File: lib/zuora/invoice_item.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module Zuora
+  module Transactions
+    class InvoiceItem < Resource
+      class << self
+        def find_by_invoice_id(invoice_id)
```
Comment (ryancyq) [inline lib/zuora/invoice_item.rb]: Don't think this is correct. Using the get API means that we are storing a collection into the single resource `InvoiceItem`

Comment (ryancyq) [reply lib/zuora/invoice_item.rb]: It should just be `Zuora::Transaction::Invoice.invoice_items` that returns a collection of `zuora::Transaction::InvoiceItem` instances

**Pull request merged**